docs/advanced_topics.rst

```diff
@@ -0,0 +1,43 @@
+Advanced topics
+=================
+
+In this section, you can find some advanced functionalities and topics that are just partially mentioned in the API description. For the typical use case, DALI default configuration should perform well out of the box and you don't need to worry about those.
+
+DALI internal data format
+-------------------------
+
+Internally, DALI processes data in NHWC format. To reshape it to NCHW please use ``\*Permute`` operator family. It also means that if mentioned operators are used in the middle of the pipeline then next operations will take the wrong assumption about processed data format and the result is undefined.
+
+Thread affinity
+---------------
+
+The rationale behind this functionality it to allow pining DALI threads to given CPU. It can save the overhead of worker threads jumping from core to core and increase performance for CPU heave workloads.
+It is possible to set CPU thread affinity for DALI using ``DALI_AFFINITY_MASK`` environment variable. ``DALI_AFFINITY_MASK`` should be a comma separated list of CPU ids numbers that will be assigned to corresponding DALI threads. A number of DALI threads is set during pipeline construction by ``num_threads`` argument, ``set_affinity`` enables setting thread affinity for the CPU worker threads. It is worth to mention that due to performance reason nvJPEG decoder operator houses threads on its own and they are always affined.
+When ``DALI_AFFINITY_MASK`` is empty affinity is set based on the result returned by nvmlDeviceGetCpuAffinity.
+If there are more threads than values in ``DALI_AFFINITY_MASK`` value from nvmlDeviceGetCpuAffinity will be used for a thread without value provided by an environment variable,
+I.e.:
+
+.. code-block:: bash
+
+  num_threads=5
+  DALI_AFFINITY_MASK=3,5,6,10
+
+Will set thread 0 to CPU 3, 1 to CPU 5, 2 to CPU 6, 3 to CPU 10 and 4 to CPU which is returned by nvmlDeviceGetCpuAffinity.
+
```

Comment:
If the number of threads are more than the CPU id numbers in the environment variable ``DALI_AFFINITY_MASK``, then the following applies: First, the threads are assigned to CPU id numbers until all CPU id numbers from ``DALI_AFFINITY_MASK`` are assigned. Then, for the remaining threads, the CPU id numbers from `nvmlDeviceGetCpuAffinity` will be used. See below example:

.. code-block:: bash
  num_threads=5
  DALI_AFFINITY_MASK=3,5,6,10
this will set thread 0 to CPU 3, thread 1 to CPU 5, thread 2 to CPU 6, thread 3 to CPU 10 and thread 4 to CPU id that is returned by `nvmlDeviceGetCpuAffinity.`

Reply:
Adjusted